crates/compiler/builtins/roc/Hash.roc

```diff
@@ -0,0 +1,68 @@
+interface Hash
+    exposes [
+        Hash,
+        Hasher,
+        hash,
+        addBytes,
+        addU8,
+        addU16,
+        addU32,
+        addU64,
+        addU128,
+        addI8,
+        addI16,
+        addI32,
+        addI64,
+        addI128,
+        complete,
+    ] imports []
+
+## A value that can hashed.
+Hash has
+    ## Hashes a value into a [Hasher].
+    ## Note that [hash] does not produce a hash value itself; the hasher must be
+    ## [complete]d in order to extract the hash value.
+    hash : a, hasher -> hasher | a has Hash, hasher has Hasher
+
+## Describes a hashing algorithm that is fed bytes and produces an integer hash.
+##
+## The [Hasher] ability describes general-purpose hashers. It only allows
+## emission of 64-bit unsigned integer hashes. It is not suitable for
+## cryptographically-secure hashing.
+Hasher has
```

Comment:
So 2 questions:

1. Should this have some method to create the hasher or is that always up to the individual hash impl to create?
2. What about seeding of the hash function? I realized that we haven't discussed that. (maybe it is more of an issue for when I impl wyhash in the roc standard lib) @rtfeldman any thought around this? Just use a static seed/make it optional since we have other plans for hash flooding defense?

Reply:
I think the individual hash impls should expose their own functions to create the hasher (and the seed should be decided there too). That way they can expose custom options and whatever else they want. This ability is just to have a common interface to use the hasher.

Otherwise we would need to enforce a common options interface or support ability-associated datatypes, neither of which sound appealing.